tests/integration/l1l3Bridger.test.ts

```diff
@@ -0,0 +1,556 @@
+import { config, testSetup } from '../../scripts/testSetup'
+import { Address, Erc20L1L3Bridger, L1ToL2MessageStatus } from '../../src'
+import { L2ForwarderContractsDeployer__factory } from '../../src/lib/abi/factories/L2ForwarderContractsDeployer__factory'
+import { TestERC20__factory } from '../../src/lib/abi/factories/TestERC20__factory'
+import { TestERC20 } from '../../src/lib/abi/TestERC20'
+import { L1Teleporter__factory } from '../../src/lib/abi/factories/L1Teleporter__factory'
+import { fundL1, fundL2, skipIfMainnet } from './testHelpers'
+import { BigNumber, ethers } from 'ethers'
+import {
+  EthL1L3Bridger,
+  RelayedErc20L1L3Bridger,
+} from '../../src/lib/assetBridger/l1l3Bridger'
+import { expect } from 'chai'
+
+type Unwrap<T> = T extends Promise<infer U> ? U : T
+
+function poll(
+  fn: () => Promise<boolean>,
+  pollInterval: number
+): Promise<boolean> {
+  return new Promise((resolve, reject) => {
+    const interval = setInterval(async () => {
+      try {
+        const result = await fn()
+        if (result === true) {
+          clearInterval(interval)
+          resolve(true)
+        }
+      } catch (e) {
+        clearInterval(interval)
+        reject(e)
+      }
+    }, pollInterval)
+  })
+}
+
+describe('L1 to L3 Bridging', () => {
+  let setup: Unwrap<ReturnType<typeof testSetup>>
+  const l2JsonRpcProvider = new ethers.providers.JsonRpcProvider(config.arbUrl)
+
+  // setup for all test cases
+  before(async function () {
+    await skipIfMainnet(this)
+
+    setup = await testSetup()
+
+    // fund signers on L1 and L2
+    await fundL1(setup.l1Signer, ethers.utils.parseEther('1'))
+    await fundL2(setup.l2Signer, ethers.utils.parseEther('1'))
+  })
+
+  describe('ETH Bridging', () => {
+    describe('EthL1L3Bridger', () => {
+      let l1l3Bridger: EthL1L3Bridger
+
+      before(() => {
+        l1l3Bridger = new EthL1L3Bridger(setup.l3Network)
+      })
+
+      // send some eth to L3 with custom l3 recipient and l2 refund address
+      // makes sure that appropriate amounts land at the right places
+      it('happy path', async () => {
+        const l3Recipient = ethers.utils.hexlify(ethers.utils.randomBytes(20))
+        const l2RefundAddress = ethers.utils.hexlify(
+          ethers.utils.randomBytes(20)
+        )
+
+        const depositTx = await l1l3Bridger.deposit(
+          {
+            amount: ethers.utils.parseEther('0.1'),
+            to: l3Recipient,
+            l2RefundAddress: l2RefundAddress,
+          },
+          setup.l1Signer,
+          setup.l2Signer.provider!,
+          setup.l3Signer.provider!
+        )
+
+        const depositReceipt = await depositTx.wait()
+
+        // poll status
+        await poll(async () => {
```

Comment:
Is it possible to something like below instead of manually polling?
```
const depositTx = await l1l3Bridger.deposit(...);
const depositReceipt = await depositTx.wait()
const l1tol2Message = depositReceipt.getL1ToL2Messages()[0]
const l2TxReceipt = new L1TransactionReceipt(await l1ToL2Message.waitForL2())
const l2tol3Message = depositReceipt.getL1ToL2Messages()[0]
await l2toL3Message.waitForL2()
```

Reply:
i had a function like this before instead of `getDepositStatus` but the full stack team said they wouldn't use it and would instead do polling. I thought it might be best to just take it out

Reply:
Oh, fair enough.

Here though I mean that we could just do the above code instead of polling to make your test simpler.